roadmap/implementers-guide/src/node/backing/statement-distribution.md

```diff
@@ -1,107 +1,449 @@
+
+- `dispatch_requests`
+  - Dispatches pending requests for candidate data & statements.
+- `answer_request`
+  - Answers an incoming request for a candidate.
+  - Takes an incoming `AttestedCandidateRequest`.
+- `receive_response`
+  - Wait on the next incoming response.
+  - If there are no requests pending, this future never resolves.
+  - Returns `UnhandledResponse`
+- `handle_response`
+  - Handles an incoming response.
+  - Takes `UnhandledResponse`
+
+## Manifests
+
+A manifest is a message about a known backed candidate, along with a description
+of the statements backing it. It can be one of two kinds:
+
+- `Full`: Contains information about the candidate and should be sent to peers
+  who may not have the candidate yet.
+- `Acknowledgement`: Omits information implicit in the candidate, and should be
+  sent to peers which are guaranteed to have the candidate already.
+
+### Manifest Exchange
+
+Manifest exchange is when a receiving node received a `Full` manifest and
+replied with an `Acknowledgement`. It indicates that both nodes know the
+candidate as valid and backed. This allows the nodes to send `Statement`
+messages directly to each other for any new statements.
+
+Why? This limits the amount of statements we'd have to deal with w.r.t.
+candidates that don't really exist. Limiting out-of-group statement distribution
+between peers to only candidates that both peers agree are backed and exist,
+ensures we only have to store statements about real candidates.
+
+In practice, manifest exchange means that one of three things have happened:
+
+- They announced, we acknowledged
+- We announced, they acknowledged
+- We announced, they announced (not sure if this can actually happen; it would
+  happen if 2 nodes had each other in their sending set and they sent manifests
+  at the same time. The code accounts for this anyway)
+
+After the exchange is complete, we update pending statements. Pending statements
+are those we know locally that the remote node does not.
+
+#### Alternative Paths Through The Topology
+
+Nodes should send a `BackedCandidateAcknowledgement(CandidateHash,
+StatementFilter)` notification to any peer which has sent a manifest, and the
+candidate has been acquired by other means. This keeps alternative paths through
+the topology open, which allows nodes to receive additional statements that come
+later, but not after the candidate has been posted on-chain.
+
+This is mostly about the limitation that the runtime has no way for block
+authors to post statements that come after the parablock is posted on-chain and
+ensure those validators still get rewarded. Technically, we only need enough
+statements to back the candidate and the manifest + request will provide that.
+But more statements might come shortly afterwards, and we want those to end up
+on-chain as well to ensure all validators in the group are rewarded.
+
+For clarity, here is the full timeline:
+
+1. candidate seconded
+1. distributed along grid
```

Comment:
a candidate wouldn't be distributed along the grid until it is backable in the cluster